src/renderer/src/routes/pages/BuyAndSellTokens/index.tsx

```diff
@@ -0,0 +1,123 @@
+import { Dispatch, Fragment, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { TbShoppingBag } from 'react-icons/tb'
+import { Location, useLocation, useNavigate } from 'react-router-dom'
+import { HelpButton } from '@renderer/components/HelpButton'
+import { useMountUnsafe } from '@renderer/hooks/useMount'
+import { ContentLayout } from '@renderer/layouts/ContentLayout'
+import { MainLayout } from '@renderer/layouts/Main'
+import { TTokenBalance } from '@shared/@types/query'
+import { IAccountState } from '@shared/@types/store'
+
+import { BuyTokensContent } from './BuyTokensContent'
+import { isConfigured } from './constants'
+import { SellTokensContent } from './SellTokensContent'
+
+export enum BuyAndSellTokensScreenType {
+  BUY_TOKENS = 'buy-tokens',
+  SELL_TOKENS = 'sell-tokens',
+}
+
+export type TDepositActionsData = {
+  amount: string
+  address: string
+  isFeeLoading: boolean
+  fee?: string
+  token?: TTokenBalance
+  account?: IAccountState
+}
+
+type TBuyAndSellTokensContentProps = {
+  depositActionsData: TDepositActionsData | null
+  setDepositActionsData: Dispatch<TDepositActionsData | null>
+  screenType: BuyAndSellTokensScreenType
+  setScreenType: Dispatch<BuyAndSellTokensScreenType>
+  account?: IAccountState
+}
+
+const BuyAndSellTokensContent = ({
+  account,
+  depositActionsData,
+  setDepositActionsData,
+  screenType,
+  setScreenType,
+}: TBuyAndSellTokensContentProps) => {
+  const { t } = useTranslation('pages', { keyPrefix: 'buyAndSellTokens' })
+
+  if (!isConfigured)
+    return <h2 className="text-center bg-gray-700/60 py-14 px-4 rounded text-white text-2xl">{t('notConfigured')}</h2>
+
+  return (
+    <Fragment>
+      <BuyTokensContent
+        hidden={screenType !== BuyAndSellTokensScreenType.BUY_TOKENS}
+        account={account}
+        setScreenType={setScreenType}
+      />
+
+      <SellTokensContent
+        hidden={screenType !== BuyAndSellTokensScreenType.SELL_TOKENS}
+        account={account}
+        depositActionsData={depositActionsData}
+        setDepositActionsData={setDepositActionsData}
+        setScreenType={setScreenType}
+      />
+    </Fragment>
+  )
+}
+
+const BuyAndSellTokensRightActions = (
+  <div className="flex gap-x-2">
+    <HelpButton />
+  </div>
+)
+
+type TLocationState = {
+  account?: IAccountState
+}
+
+export const BuyAndSellTokensPage = () => {
+  const { t } = useTranslation('pages', { keyPrefix: 'buyAndSellTokens' })
+  const { state } = useLocation() as Location<TLocationState>
+  const navigate = useNavigate()
+  const [screenType, setScreenType] = useState(BuyAndSellTokensScreenType.BUY_TOKENS)
+  const [depositActionsData, setDepositActionsData] = useState<TDepositActionsData | null>(null)
+
+  const account = state?.account
+
+  const handleBackClick = () => {
+    navigate(`/app/wallets/${account!.id}/overview`)
+  }
+
+  useMountUnsafe(() => {
+    return () => {
+      setDepositActionsData(null)
+    }
+  })
```

Comment:
Is this necessary ? The state will already be destroyed when the component unmounts